addon/-private/system/diff-array.ts

```diff
@@ -10,7 +16,7 @@
       removedCount: <integer>       // 0 if no change
     }
 */
-export default function diffArray(oldArray, newArray) {
+export default function diffArray<T = any>(oldArray: T[], newArray: T[]): ArrayDiffResult {
```

Comment:
in what scenario will `T` fail to be inferred?

Reply:
Please remove the `= any`. There's no need for a type parameter default here.

Reply:
There's no need for the `= any` here.

Reply:
it is removed